### ISSUE TITLE
Facebook login stop working after updating to 0.5.2 

### ISSUE THREAD

**Issue by cutiko** [firebase/FirebaseUI-Android#290]
Greetings

I havent being arround for a while, but I notice there is a new version. The problem is, Facebook login keep crashing.

```
09-07 21:10:40.107 28023-28023/cl.cutiko.flash E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     Process: cl.cutiko.flash, PID: 28023
                                                                     java.lang.UnsupportedOperationException
                                                                         at java.util.AbstractList.add(AbstractList.java:404)
                                                                         at java.util.AbstractList.add(AbstractList.java:425)
                                                                         at com.firebase.ui.auth.provider.FacebookProvider.startLogin(FacebookProvider.java:90)
                                                                         at com.firebase.ui.auth.ui.idp.AuthMethodPickerActivity$1.onClick(AuthMethodPickerActivity.java:127)
                                                                         at android.view.View.performClick(View.java:5198)
                                                                         at android.view.View$PerformClick.run(View.java:21147)
                                                                         at android.os.Handler.handleCallback(Handler.java:739)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                         at android.os.Looper.loop(Looper.java:148)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                         at java.lang.reflect.Method.invoke(Native Method)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

At first I thought it could be something related to the OS so switch from API 19 (Samsung galaxy note 3 neo) to API 21 (Sony Xperia), both physical devices.

Im doing exactly what is done in the [documentation] (https://github.com/firebase/FirebaseUI-Android/blob/master/auth/README.md)

```
startActivityForResult(
                    AuthUI.getInstance()
                            .createSignInIntentBuilder()
                            .setProviders(
                                    AuthUI.EMAIL_PROVIDER,
                                    AuthUI.GOOGLE_PROVIDER,
                                    AuthUI.FACEBOOK_PROVIDER)
                            .build(),
                    RC_SIGN_IN);
```

**Comment by MobilefactoryAT** [comment]
I had to add the facebook permission xml than it worked

**Comment by cutiko** [comment]
Greetings

As explained by @MobilefactoryAT the solution is to add the `facebook_permissions` array in the strings.xml file

```
    <array name="facebook_permissions">
        <item>email</item>
        <item>public_profile</item>
    </array>
```

By other hand adding an empty array, wont solve the problem, need to have the two default items.

@samtstern the documentations says "If you would like to override these scopes", that conditional "if", would not apply. Can I suggest moving that part next to where the "facebook_application_id" is explained in the [top of the document](https://github.com/firebase/FirebaseUI-Android/blob/master/auth/README.md#identity-provider-configuration), while enforcing to do it mandatorily. Thanks.

**Comment by samtstern** [comment]
@cutiko this is a bug (and it's my fault), you can do one of two things as a workaround:

- Add the `array` in XML like you said.
- Downgrade to `0.5.1`

**Comment by samtstern** [comment]
This issue is fixed in version `0.5.3`

**Comment by cutiko** [comment]
Thanks for the fix, just came here to confirm is working, just test it and login was successful without array
